packages/database/src/migrations/20210123065802-MigrateRadioQuestionsToBinaryQuestionsInLaosSchools-modifies-data.js

```diff
@@ -0,0 +1,170 @@
+ * 'SchCVD016b',
+ * 'SchCVD016c',
+ * 'SchCVD017',
+ * 'SchCVD017a',
+ * 'SchCVD017b',
+ * 'SchCVD017c',
+ * 'SchCVD017d',
+ * 'SchCVD019',
+ * 'SchCVD020',
+ * 'SchCVD022',
+ * 'SchCVD022l',
+ * 'SchCVD024',
+ * 'SchCVD028',
+ * 'SchDP_AEAL',
+ * 'SchDP_CRS',
+ * 'SchDP_HII',
+ * 'SchDP_Plan',
+ * 'SchDP_RtR',
+ * 'SchDP_UNICEF',
+ * 'SchDP_WB',
+ * 'SchDP_WC',
+ * 'SchDP_WFP',
+ * 'SchDP_WR',
+ * 'SchDP_WV',
+ * 'SchFF001',
+ * 'SchFF002',
+ * 'SchFF004',
+ * 'SchFF011',
+ * 'SchQuar001'
+ */
+
+// List of surveys that the questions aboved are included (All Laos Schools surveys)
+/**
+ * 'Laos O2 Survey',
+ * 'School COVID-19 Response DL',
+ * 'School COVID-19 Response Laos',
+ * 'School Dev Partners Laos',
+ * 'School Electricity Laos',
+ * 'School Fundamentals DL (obsolete)',
+ * 'School Fundamentals Laos (obsolete)',
+ * 'School Quarantine Laos',
+ * 'School WASH Laos',
+ * 'School Water Supply Laos'
+ */
+
+const selectLaosSchoolsRadioQuestionsOptions = async db =>
+  db.runSql(`
+    select distinct question.code, question.options
+    from question 
+    inner join survey_screen_component on survey_screen_component.question_id = question.id
+    inner join survey_screen on survey_screen.id = survey_screen_component.screen_id
+    inner join survey on survey.id = survey_screen.survey_id
+    where question.type = 'Radio'
+    and survey.country_ids::text = '{${LAOS_SCHOOLS_COUNTRY_ID}}'
+    `);
+
+const updateQuestion = async (db, code, newOptions) =>
+  db.runSql(`
+    update question
+    set options = ARRAY[${arrayToDbString(newOptions)}],
+    type = 'Binary'
+    where code = '${code}';
+`);
+
+const isYesNoRadioQuestion = options => {
+  if (options.length !== 2) {
+    return false;
+  }
+
+  const [firstOption, secondOption] = options;
+
+  if (
+    ((firstOption.value === 'Yes' && secondOption.value === 'No') ||
+      (firstOption.value === 'No' && secondOption.value === 'Yes')) &&
+    firstOption.label &&
+    secondOption.label
+  ) {
+    return true;
+  }
+
+  return false;
+};
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+exports.up = async function (db) {
+  const { rows } = await selectLaosSchoolsRadioQuestionsOptions(db);
+
+  for (const row of rows) {
+    const { code, options: optionStringArray } = row;
+    let options;
+
+    try {
+      options = optionStringArray.map(optionString => JSON.parse(optionString));
+    } catch (e) {
+      // option is not a JSON string, skip
+      continue;
+    }
+
+    if (isYesNoRadioQuestion(options)) {
+      const newOptions = options.map(({ label, color }) => {
+        let newOption = { label };
+        if (color) {
+          newOption = {
+            ...newOption,
+            color,
+          };
+        }
+
+        return JSON.stringify(newOption);
```

Comment:
If you make the change suggested [here](https://github.com/beyondessential/tupaia/pull/1939#discussion_r563423956) this could also be simpler (just stringify the whole object, value included)